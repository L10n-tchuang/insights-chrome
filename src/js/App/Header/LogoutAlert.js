import React from 'react';
import { Alert, AlertActionCloseButton } from '@patternfly/react-core';
import cookie from 'js-cookie';

class LogoutAlert extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            alertOneVisible: cookie.get('cs_loggedOut') === 'true',
            delay: 5000
        };
    }

    hideAlertOne = () => {
        this.setState({ alertOneVisible: false });
        cookie.set('cs_loggedOut', 'false');
    }
    componentDidMount() {
        this.setTimer();
    }

    setTimer = () => {
        if (this._timer !== null) {
            clearTimeout(this._timer);
        }

        // hide after `delay` milliseconds
        this._timer = setTimeout(function() {
            this.setState({ alertOneVisible: false });
            this._timer = null;
        }.bind(this), this.state.delay);
    }

    render() {
        const { alertOneVisible } = this.state;
        return (
            <React.Fragment>
                { alertOneVisible && (
                    <Alert
                        variant="success"
                        title="You have successfully logged out."
                        className="ins-c-alert__logout"
                        action={ <AlertActionCloseButton onClose={ this.hideAlertOne } /> }
                    >
                    </Alert>
                ) }
            </React.Fragment>
        );
    }
}

export default LogoutAlert;
