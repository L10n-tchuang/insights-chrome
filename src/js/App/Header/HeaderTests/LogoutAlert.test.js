import React from 'react';
import { render } from '@testing-library/react';
import LogoutAlert from '../LogoutAlert';

describe('Login', () => {
  it('should render correctly with no content', () => {
    Object.defineProperty(window.document, 'cookie', {
      writable: true,
      value: `cs_loggedOut=false`,
    });
    const { container } = render(<LogoutAlert />);
    expect(container.querySelector('div')).toMatchSnapshot();
  });

  it('should render correctly with content', () => {
    Object.defineProperty(window.document, 'cookie', {
      writable: true,
      value: `cs_loggedOut=true`,
    });
    const mockClose = jest.fn();
    const { container } = render(<LogoutAlert onClose={mockClose} />);
    expect(container.querySelector('div')).toMatchSnapshot();
  });
});
