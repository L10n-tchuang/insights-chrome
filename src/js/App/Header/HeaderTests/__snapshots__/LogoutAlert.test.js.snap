// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly with content 1`] = `
<div
  action="[object Object]"
  aria-label="Success Alert"
  class="pf-c-alert pf-m-success ins-c-alert__logout"
  data-ouia-component-id="0"
  data-ouia-component-type="PF4/Alert"
  data-ouia-safe="true"
>
  <div
    class="pf-c-alert__icon"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
      />
    </svg>
  </div>
  <h4
    class="pf-c-alert__title"
  >
    <span
      class="pf-u-screen-reader"
    >
      Success alert:
    </span>
    You have successfully logged out.
  </h4>
  
</div>
`;

exports[`Login should render correctly with no content 1`] = `null`;
